Access is denied

When running componentize-py 0.4.0 on Windows for the [demo](https://github.com/dicej/component-sandbox-demo) I get the following error:

```
AssertionError: Access is denied. (os error 5)
```

Any ideas on where this error is coming from? It's extremely vague and I'm not sure how to determine which part of the code is generating this error

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "componentize-py"
-version = "0.4.1"
+version = "0.4.2"
 edition = "2021"
 exclude = ["cpython"]
 
@@ -35,6 +35,7 @@ futures = "0.3.28"
 tokio = { version = "1.27.0", features = ["macros", "rt", "rt-multi-thread", "fs"] }
 bytes = "1.4.0"
 pretty_env_logger = "0.5.0"
+cap-std = "2.0.0"
 
 [dev-dependencies]
 async-trait = "0.1.68"

--- a/README.md
+++ b/README.md
@@ -37,7 +37,7 @@ we'll switch.  Specifically, the remaining patches are:
 ### Prerequisites
 
 - Tools needed to build [CPython](https://github.com/python/cpython) (Make, Clang, etc.)
-- [Rust](https://rustup.rs/) stable 1.68 or later *and* nightly 2023-07-27 or later, including the `wasm32-wasi` and `wasm32-unknown-unknown` targets
+- [Rust](https://rustup.rs/) stable 1.71 or later *and* nightly 2023-07-27 or later, including the `wasm32-wasi` and `wasm32-unknown-unknown` targets
   - Note that we currently use the `-Z build-std` Cargo option to build the `componentize-py` runtime with position-independent code (which is not the default for `wasm32-wasi`) and this requires using a recent nightly build of Rust.
   
 For Rust, something like this should work once you have `rustup`:

--- a/examples/http/README.md
+++ b/examples/http/README.md
@@ -14,7 +14,7 @@ which may differ from later revisions.
 ## Prerequisites
 
 * `dicej/spin` branch `wasi-http`
-* `componentize-py` 0.4.1
+* `componentize-py` 0.4.2
 * `Rust`, for installing `Spin`
 
 ```

--- a/examples/matrix-math/README.md
+++ b/examples/matrix-math/README.md
@@ -12,7 +12,7 @@ within a guest component.
 ## Prerequisites
 
 * `wasmtime-py` 13 or later
-* `componentize-py` 0.4.1
+* `componentize-py` 0.4.2
 * `NumPy`, built for WASI
 
 Note that we must build `wasmtime-py` from source until version 13 has been

--- a/examples/matrix-math/host.py
+++ b/examples/matrix-math/host.py
@@ -30,9 +30,10 @@ matrix_math = Root(
     store,
     RootImports(
         host=Host(),
+        environment=HostEnvironment(),
         # As of this writing, `wasmtime-py` does not yet support WASI Preview 2,
         # and our example won't use it at runtime anyway, so we provide `None`
-        # for all `wasi-cli` interfaces:
+        # for most `wasi-cli` interfaces:
         poll=None,
         monotonic_clock=None,
         wall_clock=None,
@@ -40,13 +41,12 @@ matrix_math = Root(
         types=None,
         preopens=None,
         random=None,
-        environment=HostEnvironment(),
         exit=None,
         stdin=None,
         stdout=None,
         stderr=None,
         terminal_input=None,
-        terminal_output=None,        
+        terminal_output=None,
         terminal_stdin=None,
         terminal_stdout=None,
         terminal_stderr=None
@@ -59,4 +59,3 @@ if isinstance(result, Ok):
     print(f"result: {result.value}")
 else:
     print(f"eval error: {result.value}")
-

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -7,7 +7,7 @@ features = ["pyo3/extension-module"]
 
 [project]
 name = "componentize-py"
-version = "0.4.1"
+version = "0.4.2"
 description = "Tool to package Python applications as WebAssembly components"
 readme = "README.md"
 license = { file = "LICENSE" }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -319,7 +319,8 @@ pub async fn componentize(
         .env("COMPONENTIZE_PY_APP_NAME", app_name)
         .env("PYTHONHOME", "/python")
         .preopened_dir(
-            Dir::from_std_file(File::open(stdlib.path())?),
+            Dir::open_ambient_dir(stdlib.path(), cap_std::ambient_authority())
+                .with_context(|| format!("unable to open {}", stdlib.path().display()))?,
             DirPerms::all(),
             FilePerms::all(),
             "python",
@@ -327,7 +328,8 @@ pub async fn componentize(
 
     for (index, path) in python_path.iter().enumerate() {
         wasi.preopened_dir(
-            Dir::from_std_file(File::open(path)?),
+            Dir::open_ambient_dir(path, cap_std::ambient_authority())
+                .with_context(|| format!("unable to open {path}"))?,
             DirPerms::all(),
             FilePerms::all(),
             &index.to_string(),

